GameServer/Logic/GameObjects/Spell.cs

```diff
@@ -364,24 +364,31 @@ public virtual void update(long diff)
          * In here we apply the effects : damage, buffs, debuffs...
          */
 
-        public void applyEffects(Unit u, Projectile p = null)
+        public void applyEffects(Unit u, Projectile p = null, Laser l = null)
```

Comment:
Do we need to separate `Projectile` and `Laser`, or could they perhaps share a common subclass to avoid duplicate code?